src/com/lftechnology/java/training/srijan/basics/coreapp/dao/EmployeeDaoImpl.java

```diff
@@ -0,0 +1,390 @@
+
+import com.lftechnology.java.training.srijan.basics.coreapp.domain.Employee;
+import com.lftechnology.java.training.srijan.basics.coreapp.domain.SearchEmployee;
+import com.lftechnology.java.training.srijan.basics.coreapp.domain.UserRole;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class EmployeeDaoImpl implements EmployeeDao {
+
+	private static final Logger LOGGER =
+		Logger.getLogger(EmployeeDaoImpl.class.getName());
+
+	public void addEmployee(Employee e) {
+
+		Connection con = null;
+		con = DbConnection.dbConnection();
+		PreparedStatement prepareStatementInsertEmployeeDetails = null;
+		try {
+			if (con != null) {
+				String queryEmployee =
+					"INSERT INTO employeeDetails (username,fullname,department,address,employee_role,is_terminated,password) VALUES(?,?,?,?,?,?,?)";
+				prepareStatementInsertEmployeeDetails =
+					con.prepareStatement(queryEmployee);
+				prepareStatementInsertEmployeeDetails.setString(
+					1, e.getUserName());
+				prepareStatementInsertEmployeeDetails.setString(
+					2, e.getFullName());
+				prepareStatementInsertEmployeeDetails.setString(
+					3, e.getDepartment());
+				prepareStatementInsertEmployeeDetails.setString(
+					4, e.getAddress());
+				prepareStatementInsertEmployeeDetails.setString(
+					5, e.getUserRole().name());
+				prepareStatementInsertEmployeeDetails.setBoolean(
+					6, e.getIsTerminated());
+				prepareStatementInsertEmployeeDetails.setString(
+					7, e.getPassword());
+				prepareStatementInsertEmployeeDetails.executeUpdate();
+			}
+		}
+		catch (Exception ex) {
+			LOGGER.log(Level.INFO, "Exception{0}", ex);
+		}
+		finally {
+			try {
+				prepareStatementInsertEmployeeDetails.close();
+				con.close();
+			}
+			catch (SQLException e1) {
+				LOGGER.log(Level.INFO, "SQL Exception{0}", e1);
+			}
+		}
+	}
+
+	@Override
+	public Employee findById(int employeeIdToUpdate) {
+
+		Connection con = null;
+		ResultSet resultSet = null;
+		con = DbConnection.dbConnection();
+		PreparedStatement prepareStatementUpdateInfo = null;
+		Employee employee = new Employee();
+		try {
+			if (con != null) {
+				String query = "Select * from employeeDetails where id = ?";
+				prepareStatementUpdateInfo = con.prepareStatement(query);
+				prepareStatementUpdateInfo.setInt(1, employeeIdToUpdate);
+				resultSet = prepareStatementUpdateInfo.executeQuery();
+				while (resultSet.next()) {
+					employee.setId(resultSet.getInt("id"));
+					employee.setUserName(resultSet.getString("username"));
+					employee.setFullName(resultSet.getString("fullName"));
+					employee.setDepartment(resultSet.getString("department"));
+					employee.setAddress(resultSet.getString("address"));
+					employee.setIsTerminated(
+						resultSet.getBoolean("is_terminated"));
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "is_terminated" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)